wordpress-mobile/WordPress-iOS
# [Support Forum] Update the WP and JP app's support screen

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/20116

This PR implements a few changes and updates to the support screen design. The changes are easiest to follow visually by looking at the screenshots below. 

The code changes are easiest to follow by looking at the commits:
1. 68b367c0c5a24aa4fbcbb38b1b22031ab056bfa8 removes the check for A8c emails from the WP app. This check is only relevant to Zendesk support which is no longer part of the WP app.
2. c2192dbb08a66082f017e962779afdcace1ce68e removes Zendesk code from an execution path specific to the WP app (no longer needed now that the WP app won't show Zendesk support).
3. 7964f4d1caa7bc49f41cf775173e8f054b2b249b remove the forum from the JP app (this wasn't clear in the issues I created, apologies).
4. 3d4c4329741c3e26b17c637758dddc213615154e incorporated PR review feedback (updated to the latest design, reamed & unified copy, and localized string keys)
5. 19cfbd90b168859e414b46b3fd04dd555d4620b9 removed resolved to-do code comment
6. 84850187f30ef8d758569f37dbdd099f041e3974 removed now unused code

(There are a few commit messages where I said "Zenhub" instead of "Zendesk" 😬 )

## Screenshots
| | Feature flag OFF  | Feature flag ON  |
| - | - | - |
| WordPress | <img width="539" alt="Screenshot 2023-02-13 at 20 54 39" src="https://user-images.githubusercontent.com/1898325/218604610-287d6281-90c2-47af-9575-6be5af5e4601.png"> | <img width="539" alt="Screenshot 2023-02-13 at 20 55 30" src="https://user-images.githubusercontent.com/1898325/218604622-376d0c9b-30e5-4fe8-ab73-2e0429530f1a.png"> |
| Jetpack | <img src="https://user-images.githubusercontent.com/1898325/218605384-cf443255-53ff-4041-8e18-76b50d1ad0c3.png" alt="" width="350"> | Feature flag should have no effect on Jetpack app  |

## Test

Please note: The feature flag mentioned below is the "Provide support through a forum" one.

### Ensure the JP app does not show the .org forum
1. Install the Jetpack app
2. Verify that the forum is **not** shown (the feature flag has no effect on the JP app)

### Ensure the A8c email popup is not shown
This can happen if the app has remembered an A8c email address as the user's support email
1. Install the WordPress app
2. Go to the support screen and make sure the A8c email popup doesn't show up
3. Verify that the forum is shown 

### Design & copy check
This PR also updated the support screen to the latest designs & copy (see qk5s4q4mZsXknwhMWo2vIY-fi-3882%3A14189). Please flag anything that looks off as the designs and copy have been tweaked recently.

## Out of scope of this PR
1. There's an ongoing discussion about what to do with "WordPress Help Center" (see p1675873769529629-slack-C3EA6SMFH)
2. There's also a discussion about when to ship this change while ensuring users wishing to contact support while trying to switch to the Jetpack app can do that (see p1676313024578649-slack-C04KJRDTLF7)
I'd like to leave these topics for future PRs if needed.


## Regression Notes
1. Potential unintended areas of impact

This change has the potential to inadvertently show the wrong support options in the WP and JP apps.

7. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing as seen in the above section

8. What automated tests I added (or what prevented me from doing so)

I looked at adding a test which instantiated `SupportTableViewController` and querying its view model to verify it has the right contents for each app (WP and JP). The trouble is that the view model contains private types (that conform to `ImmuTableRow`) so it's not possible to grab the view model's rows without making many types public (e.g. `HelpRow`). Happy to hear if there's a way to do it.

PR submission checklist:
